### PR TITLE
Handle optional serverless dependencies in cli

### PR DESCRIPTION
**Pull request by kwiatekus**
**Description**

Changes proposed in this pull request:

- check which modules will be required to apply function, error on missing module only when needed

**Related issue(s)**
https://github.com/kyma-project/serverless-manager/issues/84
Depends on https://github.com/kyma-project/hydroform/pull/413

### CODE REVIEW

File: cmd/kyma/apply/function/function.go [hunk 1]
```diff
@@ -103,32 +105,45 @@ func (c *command) Run() error {
...
-	if err != nil {
-		step.Failure()
-		return err
-	}
+	operators := []operator.Operator{}
 
-	kymaAddress, err := c.kymaHostAddress()
-	if err != nil {
-		step.LogErrorf("%s\n%s", err, "Check if your cluster is available and has Kyma installed.")
+	if configuration.Subscriptions != nil {
```
Comment (pPrecel) [inline cmd/kyma/apply/function/function.go:110]: The definition of the Subscriptions field looks like this:

```go
Subscriptions []Subscription    `yaml:"subscriptions,omitempty"`
```

It does mean that in this case, we should react on non-zero value as well ( not nil but empty map ). Example here:
https://go.dev/play/p/b44TCFzbzNw

```suggestion
	if len(configuration.Subscriptions) != 0 {
```

File: cmd/kyma/apply/function/function.go [hunk 2]
```diff
@@ -103,32 +105,45 @@ func (c *command) Run() error {
...
+		}
+		operators = append(operators, operator.NewSubscriptionOperator(client.Resource(subscriptionGVR).Namespace(configuration.Namespace),
+			configuration.Name, configuration.Namespace, subscriptions...))
 	}
 
-	apiRules, err := resources.NewAPIRule(configuration, kymaAddress)
-	if err != nil {
-		step.Failure()
-		return err
+	if configuration.APIRules != nil {
```
Comment (pPrecel) [inline cmd/kyma/apply/function/function.go:128]: Same thing as for the `configuration.Subscriptions` field

```suggestion
	if len(configuration.APIRules) != 0 {
```

File: cmd/kyma/apply/function/function.go [hunk 3]
```diff
@@ -103,32 +105,45 @@ func (c *command) Run() error {
...
 	}
 
-	subscriptions, err := resources.NewSubscriptions(configuration)
-	if err != nil {
-		step.Failure()
-		return err
-	}
+	operators := []operator.Operator{}
 
-	kymaAddress, err := c.kymaHostAddress()
```
Comment (pPrecel) [inline cmd/kyma/apply/function/function.go:112]: So we totally remove this functionality? Am I right?

Comment (kwiatekus) [reply cmd/kyma/apply/function/function.go:112]: Yes. API rules do not require to pass the full runtime domain. The controller is assuming it based on kyma gateway internally

**Review by pPrecel — DISMISSED** [review]

**Review by pPrecel — APPROVED** [review]

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**